modin/pandas/indexing.py

```diff
@@ -842,7 +892,19 @@ def _compute_lookup(self, row_loc, col_loc):
                         # labels that are missing and so printing the whole indexer
                         else axis_loc
                     )
-                    raise KeyError(missing_labels)
+                    if not setitem:
+                        raise KeyError(missing_labels)
+                    missing_labels = pandas.Index(missing_labels)
+                    if axis == 0:
+                        index = self.qc.index.append(missing_labels)
+                        self.qc = self.qc.reindex(labels=index, axis=0)
+                        self.df._update_inplace(new_query_compiler=self.qc)
+                    elif axis == 1:
+                        columns = self.qc.columns.append(missing_labels)
+                        self.qc = self.qc.reindex(labels=columns, axis=1)
+                        self.df._update_inplace(new_query_compiler=self.qc)
```

Comment:
Reindexing is very heavy as it triggers full-axis concatenation, do we really need this all the time? For distributed-like `item` we can call [`query_compiler.insert_item`](https://github.com/modin-project/modin/blob/e3723a63ad06eba96e1ed1e88c1821df5a762cee/modin/core/storage_formats/base/query_compiler.py#L2983), so maybe we should distribute `item` first (if it's not yet) and then use `insert_item` instead of reindexing the frame?